src/Http/Http.Abstractions/src/Routing/RouteValueDictionary.cs

```diff
@@ -2,35 +2,54 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Collections;
-using System.Collections.Concurrent;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
-using System.Reflection.Metadata;
 using System.Runtime.CompilerServices;
+#if !COMPONENTS
+using System.Collections.Concurrent;
+using System.Reflection.Metadata;
 using Microsoft.AspNetCore.Http.Abstractions;
-using Microsoft.AspNetCore.Routing;
 using Microsoft.Extensions.Internal;
+using Microsoft.AspNetCore.Routing;
+#endif
 
+#if !COMPONENTS
 [assembly: MetadataUpdateHandler(typeof(RouteValueDictionary.MetadataUpdateHandler))]
+#endif
 
+#if COMPONENTS
+namespace Microsoft.AspNetCore.Components.Routing;
+#else
 namespace Microsoft.AspNetCore.Routing;
+#endif
 
+#if !COMPONENTS
 /// <summary>
 /// An <see cref="IDictionary{String, Object}"/> type for route values.
 /// </summary>
 [DebuggerTypeProxy(typeof(RouteValueDictionaryDebugView))]
 [DebuggerDisplay("Count = {Count}")]
 public class RouteValueDictionary : IDictionary<string, object?>, IReadOnlyDictionary<string, object?>
+#else
+[DebuggerTypeProxy(typeof(RouteValueDictionaryDebugView))]
+[DebuggerDisplay("Count = {Count}")]
+internal class RouteValueDictionary : IDictionary<string, object?>, IReadOnlyDictionary<string, object?>
+#endif
```

Comment:
Can the comments and debugger attributes be moved outside of the ifdef?

Reply:
done